server/engine/telemetry.go

```diff
@@ -27,11 +28,29 @@ func GetEvent(propertiesMap analytics.Properties) string {
 func BuildSegmentPropertiesMap(db *gorm.DB) analytics.Properties {
 
 	var propertiesMap = analytics.Properties{}
+	retriesMap := make(map[string]interface{})
+	errorsMap := make(map[string]interface{})
 	var metricData []model.Telemetry
 	db.Find(&metricData)
+	// Error Details and Number of retries will be granular at the step level hence, each error and retry will be mapped to a step
+	// in a nested JSON like format
+	// Rest of the metrics are granular at the deployment level
 	for _, data := range metricData {
-		propertiesMap[string(data.MetricName)] = fmt.Sprintf("%v", data.MetricValue)
+		fmt.Printf("%+v\n", data)
+		if data.Step == "" {
+			propertiesMap[string(data.MetricName)] = fmt.Sprintf("%v", data.MetricValue)
```

Comment:
What's the need to call `fmt.Sprintf()` here? Can't we assign the metric value directly?

Reply:
You're right we can, when I was first trying this it was giving me some weird panic where it wasn't directly able to assign data.MetricValue directly to the map. But I tried it again now and it works.